client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -43,262 +37,256 @@ export async function drag(testId: any, start: any, end: any, lasso = false) {
-  const layout = await expect(page).toMatchElement(getTestId(testId));
-  const elBox = await layout.boxModel();
+export async function clickOnCoordinate(
+  testId: string,
+  coord: Coordinate,
+  page: Page
+): Promise<void> {
+  const layout = await page.getByTestId(testId);
+  const elBox = await layout.boundingBox();
 
   if (!elBox) {
     throw Error("Layout's boxModel is not available!");
   }
 
-  const x = elBox.content[0].x + coord.x;
-  const y = elBox.content[0].y + coord.y;
+  const x = elBox.x + coord.x;
+  const y = elBox.y + coord.y;
   await page.mouse.click(x, y);
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getAllHistograms(testclass: any, testIds: any) {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  const histTestIds = testIds.map((tid: any) => `histogram-${tid}`);
-
-  // these load asynchronously, so we need to wait for each histogram individually,
-  // and they may be quite slow in some cases.
-  // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 2.
-  await waitForAllByIds(histTestIds, { timeout: 4 * 60 * 1000 });
-
-  const allHistograms = await getAllByClass(testclass);
-
-  const testIDs = await Promise.all(
-    allHistograms.map((hist) =>
-      page.evaluate((elem) => elem.dataset.testid, hist)
-    )
-  );
-
-  return testIDs.map((id) => id.replace(/^histogram-/, ""));
-}
-
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getAllCategoriesAndCounts(category: any) {
+export async function getAllCategoriesAndCounts(
+  category: string,
+  page: Page
+): Promise<[string, string][]> {
   // these load asynchronously, so we have to wait for the specific category.
-  await waitByID(`category-${category}`);
-
-  return page.$$eval(
-    `[data-testid="category-${category}"] [data-testclass='categorical-row']`,
-    (rows) =>
-      Object.fromEntries(
-        rows.map((row) => {
-          // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-          const cat = row
-            .querySelector("[data-testclass='categorical-value']")
-            .getAttribute("aria-label");
-
-          const count = (
-            row.querySelector(
-              "[data-testclass='categorical-value-count']"
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-            ) as any
-          ).innerText;
-
-          return [cat, count];
-        })
-      )
+  const categoryRows = await page
+    .getByTestId(`category-${category}`)
+    .getByTestId("categorical-row")
+    .all();
+
+  return Object.fromEntries(
```

Comment:
Discussion: would `Array.from()` work here? Feels more intuitive, given that we're working with array